x/sysproxy/sysproxy_windows.go

```diff
@@ -0,0 +1,108 @@
+//go:build windows
+
+package sysproxy
+
+import (
+	"errors"
+	"fmt"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/registry"
+)
+
+var (
+	modwininet            = windows.NewLazySystemDLL("wininet.dll")
+	procInternetSetOption = modwininet.NewProc("InternetSetOptionW")
+)
+
+const (
+	INTERNET_OPTION_SETTINGS_CHANGED = 39
+	INTERNET_OPTION_REFRESH          = 37
+)
+
+func internetSetOption(hInternet uintptr, dwOption int, lpBuffer uintptr, dwBufferLength uint32) bool {
+	ret, _, _ := procInternetSetOption.Call(
+		hInternet,
+		uintptr(dwOption),
+		lpBuffer,
+		uintptr(dwBufferLength),
+	)
+	return ret != 0
+}
+
+func resetWininetProxySettings() error {
+	result1 := internetSetOption(0, INTERNET_OPTION_SETTINGS_CHANGED, 0, 0)
+	result2 := internetSetOption(0, INTERNET_OPTION_REFRESH, 0, 0)
+
+	if result1 && result2 {
+		fmt.Println("Operation successful")
```

Comment:
If this is a library, please remove `fmt.Println`, users might not want have these lines printed to the screen.